bundles/org.openhab.binding.radiothermostat/src/main/java/org/openhab/binding/radiothermostat/internal/util/RadioThermostatScheduleJson.java

```diff
@@ -0,0 +1,155 @@
+                config.friNightCoolTime, config.friMorningCoolTemp, config.friDayCoolTemp, config.friEveningCoolTemp,
+                config.friNightCoolTemp);
+        satCool = getDaySchedule(config.satMorningCoolTime, config.satDayCoolTime, config.satEveningCoolTime,
+                config.satNightCoolTime, config.satMorningCoolTemp, config.satDayCoolTemp, config.satEveningCoolTemp,
+                config.satNightCoolTemp);
+        sunCool = getDaySchedule(config.sunMorningCoolTime, config.sunDayCoolTime, config.sunEveningCoolTime,
+                config.sunNightCoolTime, config.sunMorningCoolTemp, config.sunDayCoolTemp, config.sunEveningCoolTemp,
+                config.sunNightCoolTemp);
+    }
+
+    public String getHeatProgramJson() throws IllegalStateException {
+        return getProgramJson(monHeat, tueHeat, wedHeat, thuHeat, friHeat, satHeat, sunHeat);
+    }
+
+    public String getCoolProgramJson() throws IllegalStateException {
+        return getProgramJson(monCool, tueCool, wedCool, thuCool, friCool, satCool, sunCool);
+    }
+
+    private String getProgramJson(@Nullable String mon, @Nullable String tue, @Nullable String wed,
+            @Nullable String thu, @Nullable String fri, @Nullable String sat, @Nullable String sun)
+            throws IllegalStateException {
+        // all were null, bypass
+        if (mon == null && tue == null && wed == null && thu == null && fri == null && sat == null && sun == null) {
+            return "";
+        }
+
+        // some were null, the schedule is invalid
+        if (mon == null || tue == null || wed == null || thu == null || fri == null || sat == null || sun == null) {
+            throw new IllegalStateException();
+        }
+
+        return "{\"0\":" + mon + ",\"1\":" + tue + ",\"2\":" + wed + ",\"3\":" + thu + ",\"4\":" + fri + ",\"5\":" + sat
+                + ",\"6\":" + sun + "}";
+    }
+
+    private @Nullable String getDaySchedule(@Nullable String morningTime, @Nullable String dayTime,
+            @Nullable String eveningTime, @Nullable String nightTime, @Nullable Integer morningTemp,
+            @Nullable Integer dayTemp, @Nullable Integer eveningTemp, @Nullable Integer nightTemp) {
+        // if any null, this day schedule is not valid
+        if (morningTime == null || dayTime == null || eveningTime == null || nightTime == null || morningTemp == null
+                || dayTemp == null || eveningTemp == null || nightTemp == null) {
+            return null;
+        }
+
+        final Integer morningMin = parseMinutes(morningTime);
+        final Integer dayMin = parseMinutes(dayTime);
+        final Integer eveningMin = parseMinutes(eveningTime);
+        final Integer nightMin = parseMinutes(nightTime);
+
+        // if any of the times could not be parsed into minutes, the schedule is invalid
+        if (morningMin == null || dayMin == null || eveningMin == null || nightMin == null) {
+            return null;
+        }
```

Comment:
fixed

Reply:
I think you now lost the array size validation, `hourMin.length == 2`. Maybe you could do something like this:
```java
final String[] hourMin = timeStr.split(":");
if (hourMin.length != 2) {
    throw new IllegalArgumentException("HH:mm expected");
}
```

You could then catch `IllegalArgumentException` instead, since `NumberFormatException` is a subclass hereof.

Reply:
I was thinking that it would be caught by the pattern matching in the thing configuration. Regardless, I have implemented your suggestion.

Reply:
Oh, I didn't consider/check that, I just saw that you removed the validation in your commit. If you are sure it will be validated before getting this far, feel free to simplify.

Reply:
Let me know if you prefer to keep current solution or change it.

Reply:
OK, I simplified it back to just catching NumberFormatException.